File: BehaviorDemo/app/src/main/java/com/renyu/behaviordemo/MainActivity.java
package com.renyu.behaviordemo;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.jakewharton.rxbinding.view.RxView;

import rx.functions.Action1;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        RxView.clicks(findViewById(R.id.button_fab)).subscribe(new Action1<Void>() {
            @Override
            public void call(Void aVoid) {
                startActivity(new Intent(MainActivity.this, FloatingActionButtonActivity.class));
            }
        });

        RxView.clicks(findViewById(R.id.button_htb)).subscribe(new Action1<Void>() {
            @Override
            public void call(Void aVoid) {
                startActivity(new Intent(MainActivity.this, BackActivity.class));
            }
        });

        RxView.clicks(findViewById(R.id.button_rv)).subscribe(new Action1<Void>() {
            @Override
            public void call(Void aVoid) {
                startActivity(new Intent(MainActivity.this, CardRVActivity.class));
            }
        });
    }
}

File: GradleDemo/app/src/main/java/com/renyu/gradledemo/MainActivity.kt
package com.renyu.gradledemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.renyu.route.annotations.Destination
import com.renyu.route_runtime.Route

@Destination(url = "route://page-home", description = "首页")
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        Route.goto(this, "route://page-child?a=1&b=2")
    }
}